internal/k8s/client.go

```diff
@@ -0,0 +1,97 @@
+package k8s
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+)
+
+const (
+	saPath      = "/var/run/secrets/kubernetes.io/serviceaccount"
+	caCertPath  = saPath + "/ca.crt"
+	saTokenPath = saPath + "/token"
+	apiService  = "https://kubernetes.default.svc"
+)
+
+type K8sClient struct {
+	apiUrl string
+	token  string
+	client *http.Client
+}
+
+func NewK8sClient(apiUrl string, tlsConfig *tls.Config) (*K8sClient, error) {
```

Comment:
We already have k8sClient in pkg (https://github.com/flightctl/flightctl/blob/main/pkg/k8sclient/k8s_client.go) .  Maybe they can be unified?

Reply:
I did not notice it. Makes sense to have them unified. Fixed